### PR TITLE
feat: ConfigProvider support Transfer className and style properties

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref https://github.com/ant-design/ant-design/issues/43051

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | ConfigProvider support Transfer className and style properties           |
| 🇨🇳 Chinese | ConfigProvider 支持 Transfer 组件的 className 和 style 属性          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e023c81</samp>

This pull request adds a new feature to the `ConfigProvider` component, which allows users to customize the appearance of the `Transfer` component. It updates the `ConfigProvider`, `Transfer`, and `context` files to implement this feature, and adds tests and documentation for it.

### <samp>🤖 Generated by Copilot at e023c81</samp>

*  Add a new optional `transfer` prop to the `ConfigProvider` component and its context value, which can have `className` and `style` properties to customize the `Transfer` component ([link](https://github.com/ant-design/ant-design/pull/43063/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR90-R93), [link](https://github.com/ant-design/ant-design/pull/43063/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R139-R142), [link](https://github.com/ant-design/ant-design/pull/43063/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R230), [link](https://github.com/ant-design/ant-design/pull/43063/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R280)).

### CODE REVIEW

File: components/config-provider/__tests__/index.test.tsx [hunk 1]
```diff
@@ -198,4 +199,52 @@ describe('ConfigProvider', () => {
...
+      <ConfigProvider
+        transfer={{
+          className: 'test-className',
+        }}
+      >
+        <Transfer dataSource={mockData} />
+      </ConfigProvider>,
+    );
+
+    expect(container.querySelector('.ant-transfer.test-className')).toBeTruthy();
```
Comment (li-jia-nan) [inline components/config-provider/__tests__/index.test.tsx:222]:
```suggestion
    expect(container.querySelector('.ant-transfer')).toHaveClass('test-className');
```

File: components/config-provider/__tests__/index.test.tsx [hunk 2]
```diff
@@ -198,4 +199,52 @@ describe('ConfigProvider', () => {
...
+          style: {
+            color: 'red',
+          },
+        }}
+      >
+        <Transfer dataSource={mockData} style={{ fontSize: '16px' }} />
+      </ConfigProvider>,
+    );
+
+    expect(container.querySelector('.ant-transfer')?.getAttribute('style')).toEqual(
```
Comment (li-jia-nan) [inline components/config-provider/__tests__/index.test.tsx:246]:
```suggestion
    expect(container.querySelector('.ant-transfer')?.toHaveStyle({....});
```

File: components/transfer/index.tsx [hunk 3]
```diff
@@ -336,19 +337,25 @@ const Transfer = <RecordType extends TransferItem = TransferItem>(
...
+    transfer?.className,
     className,
     rootClassName,
     hashId,
   );
 
+  const mergedStyle = React.useMemo<React.CSSProperties>(
+    () => ({ ...transfer?.style, ...style }),
+    [transfer?.style, style],
+  );
```
Comment (MadCcc) [inline components/transfer/index.tsx:349]: 这个不用 memo 了

Comment (Yuiai01) [reply components/transfer/index.tsx]: 删除掉了~

**Review by MadCcc — APPROVED** [review]

**Pull request merged**